src/compress_graph.jl

```diff
@@ -110,24 +111,19 @@ end
 
 # Check if node is trivial because of an identity on the left (:mult or :ldiv).
 function is_trivial_left(graph, node)
-    op = graph.operations[node]
-    if (op == :lincomb)
-        return false;
-    else
-        return graph.parents[node][1] == :I &&
-            (op == :mult || op == :ldiv)
-    end
+    return graph.operations[node] ∈ [:mult, :ldiv] &&
+        graph.parents[node][1] == :I
 end
 
 # Check if node is trivial because of an identity on the right (:mult).
 function is_trivial_right(graph, node)
+    return graph.operations[node] == :mult &&
+        graph.parents[node][2] == :I
```

Comment:
A lincomb can have just one parent. Explicit use of if-statement might be needed.

Reply:
Thank you for spotting this – I've added a new test that should cover this case, but I don't think there is an issue in the case you suggest.

The `&&` operator uses short-circuit evaluation, and if `graph.operations[node]` is not `:mult`, then the second condition is not checked and the operator returns `false` immediately.